business/src/main/java/org/mskcc/cbio/portal/service/ApiService.java

```diff
@@ -341,33 +326,107 @@ else if(type.equals("frequency"))
     public List<Integer> getSampleInternalIds(String study_id) {
         return sampleMapper.getSampleInternalIdsByStudy(study_id);
     }
-    
+
+	@Transactional
+	public List<DBProfileData> getGeneticProfileData(List<String> genetic_profile_ids, List<String> genes) {
+		return getGeneticProfileData(genetic_profile_ids, genes, null, null);
+	}
+
 	@Transactional
-	public List<Serializable> getGeneticProfileData(List<String> geneticProfileStableIds, List<String> hugoGeneSymbols,
-													List<String> sampleStableIds, String sampleListStableId) {
+	public List<DBProfileData> getGeneticProfileDataBySampleList(List<String> genetic_profile_ids, List<String> genes, String sample_list_id) {
+		return getGeneticProfileData(genetic_profile_ids, genes, null, sample_list_id);
+	}
 
-		List<DBGeneticProfile> profiles = getGeneticProfiles(geneticProfileStableIds);
-		List<String> mutationProfiles = new ArrayList<>();
-		List<String> nonMutationProfiles = new ArrayList<>();
-		for (DBGeneticProfile profile : profiles) {
-			if ("MUTATION_EXTENDED".equals(profile.genetic_alteration_type)) {
-				mutationProfiles.add(profile.id);
+	@Transactional
+	public List<DBProfileData> getGeneticProfileDataBySample(List<String> genetic_profile_ids, List<String> genes, List<String> sample_ids) {
+		return getGeneticProfileData(genetic_profile_ids, genes, sample_ids, null);
+	}
+
+	@Transactional
+	public List<DBProfileData> getGeneticProfileData(List<String> genetic_profile_ids, List<String> genes, List<String> sample_ids, String sample_list_id) {
+		List<DBGeneticProfile> profiles = getGeneticProfiles(genetic_profile_ids);
+		List<String> mutation_profiles = new ArrayList<>();
+		List<String> non_mutation_profiles = new ArrayList<>();
+		for (DBGeneticProfile p : profiles) {
+			if (p.genetic_alteration_type.equals("MUTATION_EXTENDED")) {
```

Comment:
if "MUTATION_EXTENDED" is used throughout the code base (I haven't check), I'd recommend making it part of an enum